proxy/src/transport/tls/config.rs

```diff
@@ -42,6 +44,15 @@ pub struct CommonSettings {
     pub private_key: PathBuf,
 }
 
+#[derive(Clone, Debug)]
+struct PathAndHash {
+    /// The path to the file.
+    path: PathBuf,
+
+    /// The last SHA-384 digest of the file, if we have previously hashed it.
+    last_hash: RefCell<Option<Digest>>,
```

Comment:
I would have preferred not to use `RefCell` here, but it was necessary for `last_hash` to have interior mutability, as `has_changed()` can't take `&mut self`, since the closure passed to`filter_map` can't borrow the `files` variable mutably. 

If there was an equivalent of `Iterator::scan` for `Stream`s, we could do that, and pass the files vector mutably to each iteration before `filter_map`ping, but that doesn't exist currently (I assume due to Lifetime Reasons?).